app/models/user.rb

```diff
@@ -1,23 +1,33 @@
 class User < ApplicationRecord
+
   validates_presence_of :username
   validates_length_of   :username,
                         :in => MIN_USERNAME_LENGTH..MAX_USERNAME_LENGTH
   validates             :username, :uniqueness => true
   validates_format_of   :username, :with => VALID_USERNAME_CHARS
 
   validates_presence_of :password_digest
+
+  validates_length_of   :password, minimum: 6, message: INVALID_PASSWORD
   has_secure_password
 
   def custom_error_messages
     messages = []
-    if username.length == 0 || password.nil?
+
+    if username.length == 0
       messages << BLANK_REGISTER_PARAMS
     elsif !username.length.between?(MIN_USERNAME_LENGTH,MAX_USERNAME_LENGTH) || !username.match(VALID_USERNAME_CHARS)
       messages << INVALID_USERNAME
     end
-    if User.find_by(username: username)
-      messages << EXISTING_USERNAME
+
+    if password.nil?
```

Comment:
Whereas here, couldn't we do message << password.nil? ? BLANK_REGISTER_PARMS : errors.messages[:password] and condense this to s single line? I am neutral on this, just an observation.

Reply:
Good looks. We can mess with the syntax on the next push. The 'nil' the end of the ternary works just fine - just looks funny. This bit of code could be put in to ternary as well. I believe they were written a few days apart, potentially not by the same person.